pkg/apis/job.min.io/v1alpha1/types.go

```diff
@@ -0,0 +1,109 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:defaulter-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Namespaced,shortName=miniojob,singular=miniojob
+// +kubebuilder:printcolumn:name="Tenant",type=string,JSONPath=`.spec.tenant.name`
+// +kubebuilder:printcolumn:name="Phase",type=string,JSONPath=`.spec.status.phase`
+
+// MinIOJob is a top-level type. A client is created for it
+type MinIOJob struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// *Required* +
+	//
+	// The root field for the MinIOJob object.
+	Spec MinIOJobSpec `json:"spec,omitempty"`
+
+	// Status provides details of the state of the MinIOJob steps
+	// +optional
+	Status MinIOJobStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MinIOJobList is a top-level list type.
+type MinIOJobList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []MinIOJob `json:"items"`
+}
+
+// MinIOJobSpec (`spec`) defines the configuration of a MinIOJob object. +
+type MinIOJobSpec struct {
+	// *Required* +
+	//
+	// Service Account name for the jobs to run
+	ServiceAccountName string `json:"serviceAccountName"`
+
+	// *Required* +
+	//
+	// TenantRef Reference for minio Tenant to eun the jobs against
+	TenantRef TenantRef `json:"tenant"`
+
+	// Execution order of the jobs, either `parallel` or `sequential`.
+	// Defaults to `parallel` if not provided.
+	// +optional
+	Execution string `json:"execution"`
```

Comment:
we should use a type here to declare the possible two values

Reply:
Adding default and enum annotations instead.
```go
// +kubebuilder:default=parallel
// +kubebuilder:validation:Enum=parallel;sequential;
Execution string `json:"execution"`
```

This works nice because now the validation is part of the CRD

```yaml
...
              execution:
                default: parallel
                enum:
                - parallel
                - sequential
                type: string
...
```

Reply:
still use the type so when using this CRD programatically we can use the type for valid values, for example `ImagePullPolicy` is a type https://github.com/minio/operator/blob/master/pkg/apis/minio.min.io/v2/types.go#L302 and when we use it we use `corev1.PullIfNotPresent` for a value for example

Reply:
![Uploading Screenshot 2023-11-22 at 7.24.56 PM.png…]()

Reply:
<img width="662" alt="Screenshot 2023-11-22 at 7 25 16 PM" src="https://github.com/minio/operator/assets/18384552/77c24d7f-bb40-41df-8c9e-c5054839d478">

Reply:
I see, added